api/app/views/spree/api/orders/could_not_transition.json.jbuilder

```diff
@@ -1,4 +0,0 @@
-# frozen_string_literal: true
-
-json.error(I18n.t(:could_not_transition, scope: "spree.api.order"))
```

Comment:
Maybe someone is using this template in their store, we should keep it, print a deprecation warning and render the generic partial.

Reply:
> Maybe we can also add some specs for other resources' controllers that have state machines? Just to be really sure it's working with any of them.

I could not find any other controller which raise an exception because of the failed state machine transition.

There are controllers that prevent any exception by controlling the flow:
[return_authorization#cancel](https://github.com/solidusio/solidus/blob/master/api/app/controllers/spree/api/return_authorizations_controller.rb#L60-L67)

[inventory_unit prepare event](https://github.com/solidusio/solidus/blob/master/api/app/controllers/spree/api/inventory_units_controller.rb#L37)

[shipment#ready](https://github.com/solidusio/solidus/blob/master/api/app/controllers/spree/api/shipments_controller.rb#L60)

They also return a different message so it will be a breaking change to make them use the generic error message.


Added deprecation for order error messages:
https://github.com/solidusio/solidus/pull/3520/files#diff-72139884662b996fcc462020939b7807R195